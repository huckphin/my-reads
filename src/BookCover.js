import React, { Component } from 'react';
import PropTypes from 'prop-types';

class BookCover extends Component {
    static propTypes = {
        height: PropTypes.number.isRequired,
        width: PropTypes.number.isRequired,
        backgroundImageURL: PropTypes.string.isRequired
    }

    render () {
        const { height, width, backgroundImageURL } = this.props.coverProps;

        return (
            <div className="book-cover" style={{ width: width, height: height, backgroundImage: `url(${backgroundImageURL})` }}>
            </div>
        );
    }
}

export default BookCover;