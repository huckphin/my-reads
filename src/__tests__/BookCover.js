import React from 'react';
import BookCover from '../BookCover';
import renderer from 'react-test-renderer';

test('render a BookCover', () => {
    const component = renderer.create(
        <BookCover
            height={ 100 }
            width= { 100 }
            backgroundImageURL="http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api"
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});