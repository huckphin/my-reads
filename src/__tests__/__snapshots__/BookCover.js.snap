// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a BookCover 1`] = `
<div
  className="book-cover"
  style={
    Object {
      "backgroundImage": "url(http://books.google.com/books/content?id=PGR2AwAAQBAJ&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73-GnPVEyb7MOCxDzOYF1PTQRuf6nCss9LMNOSWBpxBrz8Pm2_mFtWMMg_Y1dx92HT7cUoQBeSWjs3oEztBVhUeDFQX6-tWlWz1-feexS0mlJPjotcwFqAg6hBYDXuK_bkyHD-y&source=gbs_api)",
      "height": 100,
      "width": 100,
    }
  }
/>
`;
